ui/views.py

```
@@ -123,6 +123,53 @@ def create_repo(request):
     )
 
 
+def get_vocabularies(facets):
+    """
+    Parse facet information for the template.
+    It will return a dictionary that looks like this:
+
+    {
+        (u'13', u'difficulty'): [(38, u'medium', 23), (17, u'hard', 19)],
+        (u'15', u'prerequisite'): [(44, u'yes', 23)]
+    }
+
+    The keys are tuples with Vocabulary ID and name for the key,
+    and a list of tuples containing id, label, and count for the terms.
+
+    This is for ease-of-use in the template, where the integer primary keys
+    are used for search links and the names/labels are used for display.
+    """
+
+    if "fields" not in facets:
+        return {}
+    vocab_ids = []
+    term_ids = []
+    for vocab_id, counts in facets["fields"].items():
+        if not vocab_id.isdigit():
```

Wouldn't we want to raise an exception if this caused a problem?